spec/event_builder_spec.rb

```diff
@@ -34,216 +34,336 @@
     expect(conversion_event.params).to eq(@expected_conversion_params)
-    expect(conversion_event.url).to eq(@expected_conversion_url)
+    expect(conversion_event.url).to eq(@expected_event_builder_url)
     expect(conversion_event.http_verb).to eq(:post)
   end
 
   it 'should create a valid V2 Event when create_conversion_event is called with string event tag' do
-    @expected_conversion_params['eventFeatures'] = [
-      {
-        'name' => 'string_tag',
-        'type' => 'custom',
-        'value' => 'iamstring',
-        'shouldIndex' => false
-      },
-    ]
-
     event_tags = {
       'string_tag' => 'iamstring',
     }
 
+    @expected_conversion_params[:visitors][0][:snapshots][0][:events][0].merge!({
+      tags: event_tags
+    })
+
     conversion_event = @event_builder.create_conversion_event('test_event', 'test_user', nil, event_tags, {'111127' => '111128'})
     expect(conversion_event.params).to eq(@expected_conversion_params)
-    expect(conversion_event.url).to eq(@expected_conversion_url)
+    expect(conversion_event.url).to eq(@expected_event_builder_url)
     expect(conversion_event.http_verb).to eq(:post)
   end
 
   it 'should create a valid V2 Event when create_conversion_event is called with integer event tag' do
-    @expected_conversion_params['eventFeatures'] = [
-      {
-        'name' => 'integer_tag',
-        'type' => 'custom',
-        'value' => 42,
-        'shouldIndex' => false
-      },
-    ]
-
     event_tags ={
       'integer_tag' => 42,
     }
 
+    @expected_conversion_params[:visitors][0][:snapshots][0][:events][0].merge!({
+      tags: event_tags
+    })
+
     conversion_event = @event_builder.create_conversion_event('test_event', 'test_user', nil, event_tags, {'111127' => '111128'})
     expect(conversion_event.params).to eq(@expected_conversion_params)
-    expect(conversion_event.url).to eq(@expected_conversion_url)
+    expect(conversion_event.url).to eq(@expected_event_builder_url)
     expect(conversion_event.http_verb).to eq(:post)
   end
 
   it 'should create a valid V2 Event when create_conversion_event is called with float event tag' do
-    @expected_conversion_params['eventFeatures'] = [
-      {
-        'name' => 'float_tag',
-        'type' => 'custom',
-        'value' => 42.01,
-        'shouldIndex' => false
-      },
-    ]
-
     event_tags = {
       'float_tag' => 42.01,
     }
 
+    @expected_conversion_params[:visitors][0][:snapshots][0][:events][0].merge!({
+      tags: event_tags
+    })
+
+    conversion_event = @event_builder.create_conversion_event('test_event', 'test_user', nil, event_tags, {'111127'  => '111128'})
+    expect(conversion_event.params).to eq(@expected_conversion_params)
+    expect(conversion_event.url).to eq(@expected_event_builder_url)
+    expect(conversion_event.http_verb).to eq(:post)
+  end
+
+  it 'should create a valid V2 Event when create_conversion_event is called with value event tag' do
```

Comment:
Please rename to when 'create_conversion_event is called with numeric metric event tag'

Reply:
name is uniform with other unit test cases